src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -525,25 +375,26 @@ private IReadOnlyList<ITemplateInfo> TemplatesToDisplayInfoAbout
             {
                 IEnumerable<ITemplateInfo> templateList;
 
-                if (UnambiguousTemplateGroupToUse != null && UnambiguousTemplateGroupToUse.Count > 0)
+                if (_templateResolutionResult.UnambiguousTemplateGroupToUse != null && _templateResolutionResult.UnambiguousTemplateGroupToUse.Count > 0)
                 {
-                    templateList = UnambiguousTemplateGroupToUse.Select(x => x.Info);
+                    templateList = _templateResolutionResult.UnambiguousTemplateGroupToUse.Select(x => x.Info);
                 }
-                else if (!string.IsNullOrEmpty(TemplateName) && _matchedTemplatesWithSecondaryMatchInfo != null && _matchedTemplatesWithSecondaryMatchInfo.Count > 0)
+                else if (!string.IsNullOrEmpty(TemplateName) && _templateResolutionResult.MatchedTemplatesWithSecondaryMatchInfo != null && _templateResolutionResult.MatchedTemplatesWithSecondaryMatchInfo.Count > 0)
```

Comment:
My first thought on this was "yes, absolutely!". But we sometimes store "matches" in the result lists that are only partial matches, or even have some mismatch info, e.g. invalid parameters or invalid parameter values. So I'm a bit wary that the "Has" properties could be misleading... then again the lists can be misleading too. I want to think about it a bit more, but the current situation is definitely a bit kludgy.